clients/src/main/java/org/apache/kafka/common/record/AbstractRecords.java

```diff
@@ -171,6 +171,13 @@ public static int sizeInBytesUpperBound(byte magic, byte[] key, byte[] value, He
             return Records.LOG_OVERHEAD + LegacyRecord.recordSize(magic, key, value);
     }
 
+    public static int sizeInBytesUpperBound(byte magic, ByteBuffer key, ByteBuffer value, Header[] headers) {
```

Comment:
Do we really need this additional `ByteBuffer`-specific method vs just passing in its underlying `array` to the preceding method?